### PR TITLE
Dictionary for Relationships

### LINKED ISSUE #60: UMLClass Relationships Dictionary
Using a dictionary instead of a list could help with locating specific relationships in a class. 

- [ ] Change UMLClass to store a dictionary of relationships rather than a list 
- [ ] Change any function that uses relationships to work with the new dictionary 
- [ ] Add functions to UMLClass to better encapsulate the relationships data structure

Size estimate: 5
Primary: Amy
Secondary:

### PR DESCRIPTION
**Pull request by 16aburnett**
Closes #60 

I changed UMLClass to use a dictionary for relationships. It maps the related-to class name and the relationship object. 
I also added has_attribute() and has_relationship() to better encapsulate the data structures. 

I edited a lot of code to use the new UMLClass functions and dictionary. This reduced lines of code in some functions. 

I also changed the unit test file names and I edited them to work with the new dictionary. I also used the tests to ensure my changes produces the same output.

### CODE REVIEW

File: code/UMLModel.py [hunk 1]
```diff
@@ -135,20 +135,22 @@ def rename_attribute(self, class_name:str, old_attr_name:str, new_attr_name:str)
...
-        elif old_attr_name not in self.classes[class_name].attributes:
+        if not self.classes[class_name].has_attribute(old_attr_name):
             print("{} does not exist in {}.".format(old_attr_name, class_name))
+            return
         
         # checks if the inputted new attribute name already exists in the class
-        elif new_attr_name in self.classes[class_name].attributes:    
+        if self.classes[class_name].has_attribute(new_attr_name):  
             print("{} already exists in {}".format(new_attr_name, class_name))
+            return
```
Comment (16aburnett) [inline code/UMLModel.py:148]: I changed error cases to be single if statements and to return. I feel like this is cleaner than having a bunch of nested if statements that you have to sift through to find the actual modifying code.

File: code/UMLModel.py [hunk 2]
```diff
@@ -337,17 +313,18 @@ def load_model(self, filename):
...
         if not path.exists(MODEL_DIRECTORY+filename):
             print (f"{filename} does not exist")
             return
 
+        # Clear out previous model
+        self.classes = {}
+
+        # Holds the data loaded from json
+        raw_model = {}
+
```
Comment (16aburnett) [inline code/UMLModel.py:327]: I moved these lines so that the model is not cleared if the filename is rejected

File: code/UMLModel.py [hunk 3]
```diff
@@ -426,11 +403,8 @@ def list_relationships(self, class_name:str = ""):
                 if not self.classes[class_name].relationships:
                     print("Class '" + class_name + "' has no relationships")
                 else:
-                    for relationship in self.classes[class_name].relationships:
-                        if relationship.class1.name == class_name:
-                            print (class_name,"---", relationship.name, "-->",relationship.class2.name)
-                        else: 
-                            print (class_name,"---", relationship.name, "-->",relationship.class1.name)
+                    for other, relationship in self.classes[class_name].relationships.items():
+                        print (class_name,"---", relationship.name, "-->",relationship.get_other_class(class_name).name)
```
Comment (16aburnett) [inline code/UMLModel.py:407]: with get_other_class(), you do not need to determine which class is which in the relationship object.

**Review by adjusic — APPROVED** [review]
Looks great!

**Pull request merged**